nucypher/characters/lawful.py

```diff
@@ -458,12 +453,13 @@ def __init__(self,
                            crypto_power=crypto_power,
                            abort_on_learning_error=abort_on_learning_error,
                            known_nodes=known_nodes,
+                           domains=domains,
                            **character_kwargs)
 
         #
         # Self-Ursula
         #
-        if is_me is True:           # TODO: 340
+        if is_me is True:  # TODO: 340
```

Comment:
"Depends on what the meaning of the word _is_ is." :joy: 

Depends on what you're going for. If you're familiar with `===` in JavaScript, the `is` is used sorta kinda like this in this case.

In this case, `if is_me is True` will only be `True`, if and only if, `is_me` is actually a boolean `True`. If it's a truthy object, it won't be True.

I'd say it's more Pythonic to have `if is_me` and call it done here, but it does make sense here if we _only_ want `is_me` to be a boolean value. Python is weird.

Reply:
For example:
```python
a = "test"
bool(a) == True # True indeed
if a:
    print("This executes")
if a is True:
    print("This doesn't execute")
if bool(a) is True:
    print("This does execute")
```

Reply:
It depends.  They do different things.

Here, the equality variant (`if is_me:`) is correct and the code is wrong.  (At least that's how I see it).

The identity variant (`if thing is bool(...)`) is useful because you might have an object whose `__bool__` returns the proper bool type (and thus will evaluate `True` for equality), but which isn't actually the object you're testing for.

In this case, since we aren't checking for that, I assert that we're doing the wrong thing.

Reply:
The second answer to this SO question I think answers this succinctly. TL;DR: `==` is value equality and `is` is reference equality. The former is for checking the value of two objects, the latter is for checking if the two _refer_ to the same object.

https://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python

Reply:
At the end of the day: we've almost certainly got it wrong here.

Reply:
++ to `if is_me:`